tests/test_playbooks/organization_info.yml

```diff
@@ -0,0 +1,48 @@
+---
+- hosts: localhost
+  collections:
+    - theforeman.foreman
+  gather_facts: false
+  vars_files:
+    - vars/server.yml
+  tasks:
+    - include: tasks/organization.yml
+      vars:
+        organization_state: present
+
+- hosts: tests
+  collections:
+    - theforeman.foreman
+  gather_facts: false
+  vars_files:
+    - vars/server.yml
+  tasks:
+    - name: fetch organization info
+      organization_info:
+        username: "{{ foreman_username }}"
+        password: "{{ foreman_password }}"
+        server_url: "{{ foreman_server_url }}"
+        validate_certs: "{{ foreman_validate_certs }}"
+        organization: "Test Organization"
+        name: "Test Organization"
+      register: organization_info
+    - name: check organization details
+      assert:
+        that:
+          - organization_info['organization']['name'] == "Test Organization"
+          - "'parent_name' in organization_info['organization']"
+
+    - name: search organization info
+      organization_info:
+        username: "{{ foreman_username }}"
+        password: "{{ foreman_password }}"
+        server_url: "{{ foreman_server_url }}"
+        validate_certs: "{{ foreman_validate_certs }}"
+        organization: "Test Organization"
```

Comment:
```suggestion
```

same, otherwise you search for orgs with `name ~ Test` inside the org "Test Org", so you will never get any other results as "Test Org"